accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/session/impl/SessionManagerImpl.java

```diff
@@ -0,0 +1,104 @@
+package biz.netcentric.cq.tools.actool.session.impl;
+
+import biz.netcentric.cq.tools.actool.session.SessionManager;
+import org.apache.felix.scr.annotations.Activate;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Reference;
+import org.apache.felix.scr.annotations.Service;
+import org.apache.sling.api.resource.LoginException;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceResolverFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+@Service
+public class SessionManagerImpl implements SessionManager {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SessionManagerImpl.class);
+
+    ThreadLocal<Session> sessionHolder;
+
+    @Reference
+    private ResourceResolverFactory resourceResolverFactory;
+
+    @Activate
+    public void activate() {
+        sessionHolder = new ThreadLocal<>();
+    }
+
+    @Override
+    public Session getSession() throws RepositoryException {
+        Session session = getCurrentUserSession();
+        if(session != null) {
+            return session;
+        }
+
+        return getServiceUserSession(session);
+    }
+
+    @Override
+    public void close(Session session) {
+        if(session.equals(sessionHolder.get())) {
```

Comment:
maybe instead we should wrap the session in a custom implementation that prevents closing it? that way a developer cannot accidentally just call session.logout()